lib/service/user.go

```diff
@@ -121,17 +124,28 @@ func (svc *LndhubService) FindUserByLogin(ctx context.Context, login string) (*m
 	return &user, nil
 }
 
-func (svc *LndhubService) BalanceCheck(ctx context.Context, lnpayReq *lnd.LNPayReq, userId int64) (ok bool, err error) {
+func (svc *LndhubService) CheckPaymentAllowed(ctx context.Context, lnpayReq *lnd.LNPayReq, userId int64) (result *responses.ErrorResponse, err error) {
 	currentBalance, err := svc.CurrentUserBalance(ctx, userId)
 	if err != nil {
-		return false, err
+		return nil, err
 	}
 
 	minimumBalance := lnpayReq.PayReq.NumSatoshis
 	if svc.Config.FeeReserve {
 		minimumBalance += svc.CalcFeeLimit(lnpayReq.PayReq.Destination, lnpayReq.PayReq.NumSatoshis)
 	}
-	return currentBalance >= minimumBalance, nil
+	if currentBalance < minimumBalance {
+		return &responses.NotEnoughBalanceError, nil
+	}
+	volume, err := svc.GetVolumeOverPeriod(ctx, userId, time.Duration(svc.Config.MaxVolumePeriod*int64(time.Second)))
+	if err != nil {
+		return nil, err
+	}
+	if volume > svc.Config.MaxVolume {
+		sentry.CaptureMessage(fmt.Sprintf("transaction volume exceeded for user %d", userId))
```

Comment:
do we need a sentry notification there? 🤔 or is a log enough? 
but either way I think a log would be good.

Reply:
These error messages are logged in the level above.